src/markdown.cpp

```diff
@@ -1529,7 +1529,7 @@ int Markdown::Private::processLink(const std::string_view data,size_t offset)
     }
     else if ((lp=link.find('#'))!=-1 || link.find('/')!=-1 || link.find('.')!=-1)
     { // file/url link
-      if (lp!=-1 && !isURL(link))
+      if (lp!=-1 && !isURL(link) && Config_getEnum(MARKDOWN_ID_STYLE)==MARKDOWN_ID_STYLE_t::GITHUB)
```

Comment:
Looks like this still changes the behavior in case `MARKDOWN_ID_STYLE` is not `GITHUB`. Probably the expression should be:
```c++
  if (lp!=-1 && (Config_getEnum(MARKDOWN_ID_STYLE)==MARKDOWN_ID_STYLE_t::DOXYGEN) || !isURL(link)) { ... }
```

Reply:
The intended logic is if `#` character is found and either `MARKDOWN_ID_STYLE` is `DOXYGEN` or link is not a valid URL, correct? I'm asking because of missing parenthesis.

If so, then valid URLs that contain a `#` in them (for example `https://example.com/index.html#some-heading-id` will not be recognized as URLs in Doxygen style mode and instead it will attempt to link them to a section named `some-heading-id`. You can confirm this with the example files I posted.

Reply:
Maybe this is what you want:
```cpp
if (lp==0 || (lp>0 && !isURL(link) && Config_getEnum(MARKDOWN_ID_STYLE)==MARKDOWN_ID_STYLE_t::GITHUB)) { ... }
```

The left part of the `||` is the old logic and the right part is the new added logic.

Reply:
@doxygen @albert-github Pushed the change, you guys can check if it seems valid to you. Link parsing in Doxygen mode should remain unchanged with this commit, only GitHub parsing is affected now.